scripts/build_lbann.sh

```diff
@@ -772,149 +850,370 @@ if [[ -n "${INSTALL_DEPS:-}" ]]; then
+
+# Save the setup file in the build directory
+if [[ ! -e ${LBANN_BUILD_PARENT_DIR}/${LBANN_SETUP_FILE_LABEL} ]]; then
+    echo "Overwritting exising install file in ${LBANN_BUILD_PARENT_DIR}/${LBANN_SETUP_FILE_LABEL}"
+fi
+CMD="cp ${LBANN_INSTALL_FILE} ${LBANN_BUILD_PARENT_DIR}/${LBANN_SETUP_FILE_LABEL}"
+echo ${CMD} | tee -a ${LOG}
+[[ -z "${DRY_RUN:-}" ]] && { ${CMD} || warn_on_failure "${CMD}"; }
+
+
+cat > ${LBANN_INSTALL_FILE}<<EOF
+# Directory structure used for this build
+export LBANN_BUILD_LABEL=${LBANN_BUILD_LABEL}
+export LBANN_BUILD_PARENT_DIR=${LBANN_BUILD_PARENT_DIR}
+export LBANN_BUILD_DIR=${LBANN_BUILD_DIR}
+export LBANN_INSTALL_DIR=${LBANN_INSTALL_DIR}
+export LBANN_MODFILES_DIR=${LBANN_MODFILES_DIR}
+export LBANN_SETUP_FILE=${LBANN_SETUP_FILE}
+EOF
 
-    SPACK_INSTALL_ROOT=$(grep root $SPACK_ROOT/etc/spack/config.yaml | awk '{ print $2 }')
-    for ii in $(spack find --format "{prefix} {version} {name},/{hash}" |
-        grep -v -E "^(develop^master)" |
-        grep -e "${SPACK_ROOT}" -e "${SPACK_INSTALL_ROOT}" |
-        cut -f3 -d" ")
+ENV_ROOT_PKG_LIST=$(spack find -x --format "{name}")
+if [[ -n "${ENV_ROOT_PKG_LIST:-}" ]]; then
+    for p in ${ENV_ROOT_PKG_LIST}
     do
-        NAME=${ii%,*};
-        HASH=${ii#*,};
-        case ${NAME} in
-            "cuda" | "cudnn" | "ncurses" | "openssl" | "lbann")
-                echo "Skipping $ii"
-                continue
-                ;;
-        esac
-        CMD="spack buildcache check --rebuild-on-error --mirror-url file://${UPDATE_BUILDCACHE} -s ${HASH}"
-        echo -e "${NAME}:\t ${CMD}" | tee -a ${LOG}
-        if [[ -z "${DRY_RUN:-}" ]]; then
-            if ${CMD};
-            then
-                true
-            else
-                CMD="spack buildcache create -af -d ${UPDATE_BUILDCACHE} --only=package ${HASH}"
-                echo ${CMD} | tee -a ${LOG}
-                [[ -z "${DRY_RUN:-}" ]] && { ${CMD} || exit_on_failure "${CMD}"; }
-            fi
+        PKG_PYTHONPATH=$(spack build-env ${p} -- printenv PYTHONPATH)
+        if [[ -n "${PKG_PYTHONPATH}" ]]; then
+            P_ENV=$(echo "${p}" | tr '-' '_')
+cat >> ${LBANN_INSTALL_FILE}<<EOF
+# Add PYTHONPATH for top level python package: ${p}
+export ${P_ENV}_PKG_PYTHONPATH=${PKG_PYTHONPATH}
+export PYTHONPATH=\${${P_ENV}_PKG_PYTHONPATH}:\${PYTHONPATH}
+EOF
         fi
     done
-    CMD="spack buildcache update-index -d ${UPDATE_BUILDCACHE}"
+fi
+
+# if [[ -n "${SPACK_EXTRA_ROOT_PACKAGES:-}" ]]; then
+#     for p in ${SPACK_EXTRA_ROOT_PACKAGES}
+#     do
+#         PKG_PYTHONPATH=$(spack build-env ${p} -- printenv PYTHONPATH)
+#         if [[ -n "${PKG_PYTHONPATH}" ]]; then
+#             P_ENV=$(echo "${p}" | tr '-' '_')
+# cat >> ${LBANN_INSTALL_FILE}<<EOF
+# # Add PYTHONPATH for top level python package: ${p}
+# export ${P_ENV}_PKG_PYTHONPATH=${PKG_PYTHONPATH}
+# export PYTHONPATH=\${${P_ENV}_PKG_PYTHONPATH}:\${PYTHONPATH}
+# EOF
+#         fi
+#     done
+# fi
```

Comment:
```suggestion
```

Reply:
I may need this block later to avoid having to activate the spack environment.